api/src/main/java/com/messagebird/MessageBirdServiceImpl.java

```diff
@@ -253,6 +274,63 @@ public void deleteByID(String request, String id) throws UnauthorizedException,
         }
     }
 
+    /**
+     *
+     * Do get request for file from input url and stores the file in filepath.
+     * @param url     Absolute URL.
+     * @param filePath the path where the downloaded file is going to be stored.
+     * @return if it succeed, it returns filepath otherwise null or exception.
+     */
+    String doGetRequestForFileAndStore(final String url, final String filePath) throws GeneralException, UnauthorizedException, NotFoundException {
+        HttpURLConnection connection = null;
+        InputStream inputStream = null;
+
+        try {
+            connection = getConnection(url, null, METHOD_GET);
+            int status = connection.getResponseCode();
+
+            if (APIResponse.isSuccessStatus(status)) {
+                inputStream = connection.getInputStream();
+            } else {
+                inputStream = connection.getErrorStream();
```

Comment:
connection.getErrorStream() returns null, so better to throw an exception (maybe GeneralException)